### PR TITLE
[FacilityLocator] Provider Locator Address & Map Fixes

### PR DESCRIPTION
**Pull request by r-mc2**
These should remedy the the two big items we discussed this afternoon and will pair with one of @SimonSchama 's PRs going in for the API tonight.
Two items being:
- [Static Map on Details page not generating](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/15273)
- `null`s appearing in the address fields for Providers that have bad/missing/malformed data.
- (Also snuck in a couple other fixes we found along the way or were small enough to include)

## Description
Updates/Tweaks/Changes for Address & Details Map
* New mock data for testing empty address fields
* Moved LocationMap's `pinNames` out to `constants/index.js`
* Changed how to check which pin png to use in the Mapbox API call for static maps
* Additional small changes to other files to handle empty address fields, returning appropriate data to indicate to parent components, and formatting text to display to the user
* Fixed tiny visual glitch where Details page had an extra space before "Directions"
* Fixed tiny warning from ProviderDetail where it would warn about `location` not existing while being required
  * Fix was removing the `isRequired` as `location` is something mapped off of props out of the Redux store and that value isn't loaded until a service call is made in the component's `componentWillMount`

(@'s for visibility to @rtluu @ehuntdsva & @brettdt )

### CODE REVIEW

File: src/applications/facility-locator/components/search-results/LocationDirectionsLink.jsx [hunk 1]
```diff
@@ -5,15 +5,24 @@ import { buildAddressArray } from '../../utils/facilityAddress';
...
+      address = address.join(', ');
+    } else {
+      // If we don't have an address fallback on coords
+      const { lat, long } = location.attributes;
+      address = `${lat},${long}`;
+    }
 
     return (
       <span>
         <a
```
Comment (jbalboni) [inline src/applications/facility-locator/components/search-results/LocationDirectionsLink.jsx]: Not directly related to your changes: can we add `rel="noopener"` here?

File: src/applications/facility-locator/constants/index.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable camelcase */
```
Comment (jbalboni) [inline src/applications/facility-locator/constants/index.js]: Can we add this comment just surrounding the block with camel case properties?

File: src/applications/facility-locator/constants/index.js [hunk 3]
```diff
@@ -31,6 +32,17 @@ export const FacilityType = {
   VET_CENTER: 'vet_center',
 };
 
+/**
+ *
```
Comment (jbalboni) [inline src/applications/facility-locator/constants/index.js]: Should we remove this empty comment?

File: src/applications/facility-locator/utils/facilityAddress.js [hunk 4]
```diff
@@ -1,15 +1,20 @@
-import { compact } from 'lodash';
+import compact from 'lodash/compact';
+import isEmpty from 'lodash/isEmpty';
```
Comment (jbalboni) [inline src/applications/facility-locator/utils/facilityAddress.js]: For what it's worth, you can use the destructuring approach or the normal `import _ from 'lodash'` and the Lodash Babel plugin will convert the imports to look like this, so it doesn't bloat the bundle more.

**Review by jbalboni — APPROVED** [review]
Some minor comments, but LGTM

File: src/applications/facility-locator/components/search-results/LocationDirectionsLink.jsx (hunk 1 above)
Comment (r-mc2) [reply src/applications/facility-locator/components/search-results/LocationDirectionsLink.jsx]: Good call. I'd been adding that to other anchors across other portions of the Locator, too.

File: src/applications/facility-locator/constants/index.js (hunk 3 above)
Comment (r-mc2) [reply src/applications/facility-locator/constants/index.js]: Hah! I meant to actually leave a comment about this block, missed it in my rush. Thanks 👍

File: src/applications/facility-locator/utils/facilityAddress.js (hunk 4 above)
Comment (r-mc2) [reply src/applications/facility-locator/utils/facilityAddress.js]: Ahh, didn't know we had the babel plugin added. Last I paid attention we did not so I got in the habit of doing the direct imports (even mentioned it in my [overall review](https://gist.github.com/r-mc2/a2e269bc65eb77291de5cb5c20cef39f) of the app that I passed along months ago). 
I can definitely put it back to the destructure format if that's the case.

File: src/applications/facility-locator/constants/index.js (hunk 2 above)
Comment (r-mc2) [reply src/applications/facility-locator/constants/index.js]: Moved 👍

**Review by rianfowler — APPROVED** [review]

File: src/applications/facility-locator/utils/facilityAddress.js (hunk 4 above)
Comment (jbalboni) [reply src/applications/facility-locator/utils/facilityAddress.js]: I don't think we have a prescribed way of importing them, so it's up to you, I just wanted to mention that they're all equivalent from a bundle size standpoint (unless you start using `_.chain`, anyway).

**Pull request merged**